oracle/oracle.go

```diff
@@ -507,6 +507,18 @@ func NewProvider(
 	case provider.ProviderEthUniswap:
 		return provider.NewUniswapProvider(ctx, logger, endpoint, providerPairs...)
 
+	case provider.ProviderEthCamelot:
+		return provider.NewCamelotProvider(ctx, logger, endpoint, providerPairs...)
```

Comment:
Could you just have one provider struct and pass in different endpoints to determine which dex you're using? 🤔